pipeline_tools/shared/submission/create_analysis_file.py

```diff
@@ -0,0 +1,190 @@
+from pipeline_tools.shared.submission import format_map
+
+
+class AnalysisFile():
+    """AnalysisFile class implements the creation of a  json analysis file for Optimus and SS2 pipeline outputs
+
+
+    HCA system consumes json files that describe the .bam/.loom outputs of Optimus and SS2 pipelines.
+
+
+    The json files have the following form:
+
+
+        {
+        "describedBy": "https://schema.humancellatlas.org/type/file/6.2.0/analysis_file",
+        "file_core": {
+            "content_description": [],
+            "file_name": "heart_1k_test_v2_S1_L001.bam",
+            "format": "bam"
+        },
+        "provenance": {
+            "document_id": "d649938b-be5f-58cf-be04-0c1e6381eb9e",
+            "submission_date": "2021-05-24T12:00:00.000000Z"
+        },
+        "schema_type": "file"
+        }
+
+
+    See https://schema.humancellatlas.org/type/file/6.2.0/analysis_file for full spec
+    """
+
+    DCP2_MATRIX_CONTENT_DESCRIPTION = {
+        "text": "DCP/2-generated matrix",
+        "ontology": "data:3917",
+        "ontology_label": "Count Matrix",
+    }
+
+    # All analysis files will share these attributes
+    describedBy = SCHEMAS["ANALYSIS_FILE"]["describedBy"]
+    schema_type = SCHEMAS["ANALYSIS_FILE"]["schema_type"]
+    schema_version = SCHEMAS["ANALYSIS_FILE"]["schema_version"]
+
+    def __init__(
+        self,
+        input_uuid,
+        metadata_json,
+        pipeline_type,
+            workspace_version):
+
+        # Get the file version and file extension from params
+        file_extension = 'TBD'
+        file_version = 'TBD'
+        metadata_json = format_map.get_outputs(metadata_json)
+        outputs = metadata_json['outputs']
+        timestamp = format_map.convert_datetime(metadata_json['start'])
+
+        self.input_uuid = input_uuid
+        self.file_extension = file_extension
+        self.file_version = file_version
+        self.pipeline_type = pipeline_type
+        self.workspace_version = workspace_version
+        for output in outputs:
+            if '.loom' in output:
+                self.loom_output = {
+                    'provenance': {
+                        'document_id': format_map.get_uuid5(
+                            f"{str(self.input_uuid)}{os.path.splitext(outputs[output])[1]}"
+                        ),
+                        'submission_date': timestamp
+                    },
+                    'file_core': {
+                        'file_name': outputs[output].split('/')[-1],
+                        'format': format_map.get_file_format(outputs[output]),
+                        'content_description': []
+                    }
+                }
+            elif '.bam' in output:
+                self.bam_output = {
+                    'provenance': {
+                        'document_id': format_map.get_uuid5(
+                            f"{str(self.input_uuid)}{os.path.splitext(outputs[output])[1]}"
```

Comment:
The file names that need to be produced are:

87795ce9-03ce-51f3-b8d8-4ad6f8931fe0_2021-05-24T12:00:00.000000Z.json
d649938b-be5f-58cf-be04-0c1e6381eb9e_2021-05-24T12:00:00.000000Z.json

currently its:

7f384ffa-1045-5735-a10c-28da8b1810d3_2021-07-08T16:08:06.619.000000Z.json
f69fb51d-2043-5f6d-a3c2-e90f4437f547_2021-07-08T16:08:06.619.000000Z.json


please look at line 70 of create_file_descriptor.py on master to see how this is done